lib/rspec/core/filter_manager.rb

```diff
@@ -99,8 +99,10 @@ def prune_conditionally_filtered_examples(examples)
       # defined in the same file as the location filters. Excluded specs in
       # other files should still be excluded.
       def file_scoped_include?(ex_metadata, ids, locations)
-        no_location_filters = locations[ex_metadata[:absolute_file_path]].empty?
         no_id_filters = ids[ex_metadata[:rerun_file_path]].empty?
+        no_location_filters = locations[
+          File.expand_path(ex_metadata[:rerun_file_path])
+        ].empty?
```

Comment:
I have a concern here: location filtering is based on `absolute_file_path`, not `rerun_file_path` (see [this code](https://github.com/rspec/rspec-core/blob/98ade33737c1c2848aa6e32da286feef49065e8d/lib/rspec/core/metadata_filter.rb#L70)) since location filtering is based on where the example is _defined_, and it therefore seems potentially problematic to only check `rerun_file_path` here.  If we wanted to be paranoid this line could be:

``` ruby
no_location_filters = locations[
  File.expand_path(ex_metadata[:rerun_file_path])
].empty? && locations[ex_metadata[:absolute_file_path]].empty?
```

That said, I've been trying to think of an edge case that would require this extra bit of logic and I haven't been able to come up with one.  I'm not a fan of adding logic "just in case" so if we can't come up with an example that requires it to work properly I'd say we should just go with what you have here....but I figured it was worth bringing up to see if anyone can think of an example that would require that.

@JonRowe and @bentheax -- can either of you think of a case like that?